src/toolkit/matl_buy_policy.cc

```diff
@@ -153,24 +184,35 @@ std::set<RequestPortfolio<Material>::Ptr> MatlBuyPolicy::GetMatlRequests() {
   std::set<RequestPortfolio<Material>::Ptr> ports;
   bool make_req = buf_->quantity() < req_when_under_ * buf_->capacity();
   double amt;
-  
-  if (dormant_ > 0 && manager()->context()->time() % (active_ + dormant_) < active_) {
-    amt = TotalQty();
+
+  int current_time_ = manager()->context()->time();
+
+  if (never_dormant() || current_time_ < next_active_end_) {
+    // currently in the middle of active buying period
+    amt = TotalAvailable() * SampleRequestSize();
   }
-  else if (dormant_ == 0)
-    amt = TotalQty();
-  else {
-    // in dormant part of cycle, return empty portfolio
+  else if (current_time_ == next_active_end_ || current_time_ < next_dormant_end_) {
```

Comment:
The only way to get here is if `next_dormant_end_ > 0` and `next_active_end_ < current_time < next_dormant_end_ ` so I think you only need to test against `next_dormant_end_`:
```suggestion
  else if (current_time_ < next_dormant_end_) {
```